sponsors/models.py

```diff
@@ -29,8 +27,7 @@ class Sponsor(models.Model):
     year = models.IntegerField(default=2019)
     ecell_user = models.ForeignKey(
         CustomUser,
-        on_delete=models.SET_DEFAULT,
-        default=default_user.id)
+        on_delete=models.CASCADE)
```

Comment:
we don't want to lose the data added by the user if his/her account gets deleted, that's why I was setting it to the default user. This will delete all the sponsors added by the user if his account gets deleted.

Reply:
It was giving error of : CustomUser matching query doesn't exists.
And why do we need a ecell_user for this model? 
To keep track of who created it?

Reply:
Ans is this relation correct. A user can add multiple Sponsors, so shouldn't it be in CustomUser. If that's its purpose? Also there models.CASCADE would be a better fit.
As this intance is getting deleted. It makes no sense to set it to some default value. 
Tell me if i am going wrong somewhere.

Reply:
on_delete option of foreignkey works the reverse way. Models.cascade for ecell_user implies that if that if the user is deleted all the sponsors having this user as ecell_user are deleted , however deleting sponsors won't affect users in any case.

Reply:
> It was giving error of : CustomUser matching query doesn't exists.
Because that user doesn't exist
> And why do we need a ecell_user for this model?
> To keep track of who created it?
Yes

Reply:
> and is this relation correct? A user can add multiple Sponsors, so shouldn't it be in CustomUser. If that's its purpose?
Yeah the relation is correct if we place sponsors as fk in CustomUser, then that user will be able to add only one sponsor. This way each sponsor will have only one creator, but same user can add multiple sponsors.

Reply:
> It was giving error of : CustomUser matching query doesn't exists.
I think I will set it to null instead of default user, this way it won't cause any errors.